### PR TITLE
Feature/dfe 1183 deleting a file associated to a release

### PR DESCRIPTION
**Pull request by richardpopplehive**

### CODE REVIEW

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/NamingUtils.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        private static string ReplaceNonAlphaNumericWithSpaceAndTrim(string s)
+        {
+            return new string(s.Select(ReplaceNonAlphaNumericWithSpace).ToArray()).Trim();
+        }
+
+
+        private static Func<char, char> ReplaceNonAlphaNumericWithSpace =>
+            c => IsLetter(c) || IsWhiteSpace(c) || IsDigit(c) ? c : ' ';
+
+        public static string ReleaseYearTitle(string year, TimeIdentifier coverage)
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model/NamingUtils.cs]: Possible overlap here with a mechanism already written to format a TimeIdentifier and year.

See method `GovUk.Education.ExploreEducationStatistics.Data.Api.Services.TimePeriodLabelFormatter.Format`.

Each TimeIdentifier is annotated with a formatting convention.
See `TimePeriodLabelFormat` for the possible formats and a description of each of them.

Run the following test if you want examples of what it outputs (the test is currently skipped but when enabled it dumps a whole load of examples to the console)
`GovUk.Education.ExploreEducationStatistics.Data.Api.Tests.Services.TimePeriodLabelFormatterTests.Debug`

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/NamingUtilTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using Xunit;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.NamingUtils;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Model.Tests
+{
+    public class NamingUtilTests
```
Comment (benoutram) [inline src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/NamingUtilTests.cs]: You don't test any TimeIdentifiers here that are Quarters, groups of Quarters, Terms, groups of Terms, Months, EndOfMarch etc here.

Also, possible overlap with the tests in TimePeriodLabelFormatterTests. See my previous comment about the overlap between the NamingUtil and TimePeriodLabelFormatter.

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model/NamingUtils.cs (hunk 1 above)
Comment (richardpopplehive) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Model/NamingUtils.cs]: Okay - so I'm now taking the year part from the formatter - but the rest remains in the release object. There might be some mileage in future rationalising some of this further

File: src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/NamingUtilTests.cs (hunk 2 above)
Comment (richardpopplehive) [reply src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/NamingUtilTests.cs]: year stuff is now all covered with the timeformatter stuff

**Pull request merged**